sql/pg_duckdb--0.0.1.sql

```diff
@@ -1,40 +1,42 @@
 LOAD 'pg_duckdb';
 
--- If duckdb.postgres_role is configured let's make sure it's actually created.
--- If people change this setting after installing the extension they are
--- responsible for creating the user themselves. This is especially useful for
--- demo purposes and our pg_regress testing.
-DO $$
-DECLARE
-    role_name text;
-BEGIN
-    SELECT current_setting('duckdb.postgres_role') INTO role_name;
-    IF role_name != '' AND NOT EXISTS (
-      SELECT FROM pg_catalog.pg_roles
-      WHERE  rolname = role_name) THEN
-        EXECUTE 'CREATE ROLE ' || quote_ident(current_setting('duckdb.postgres_role'));
-    END IF;
-END
-$$;
+-- We create a duckdb schema to store most of our things. We explicitely
+-- don't use CREATE IF EXISTS or the schema key in the control file, so we know
+-- for sure that the extension will own the schema and thus non superusers
+-- cannot put random things in it, so we can assume it's safe. A few functions
+-- we'll put into @extschema@ so that in normal usage they get put into the
+-- public schema and are thus more easily usable. This is the case for the
+-- read_csv, read_parquet and iceberg functions. It would be sad for usability
+-- if people would have to prefix those with duckdb.read_csv
+CREATE SCHEMA duckdb;
+-- Allow users to see the objects in the duckdb schema. We'll manually revoke rights
+-- for the dangerous ones.
+GRANT USAGE ON SCHEMA duckdb to PUBLIC;
```

Comment:
We can't use `@extschema@` here?

Reply:
So it's kinda nice to group most of the `duckdb` functions in the `duckdb`, nice for auto complete and listing them basically.

Reply:
Ah sorry - I thought `@extschema@` was an alias for `duckdb`. Nevermind then!